pylint/lint/pylinter.py

```diff
@@ -938,16 +939,17 @@ def initialize(self):
             if not msg.may_be_emitted():
                 self._msgs_state[msg.msgid] = False
 
-    def check(self, files_or_modules):
+    def check(self, files_or_modules: Sequence[str]) -> None:
```

Comment:
`Union[Sequence[str], str]` maybe with a `TODO` comment after the deprecation warning to update the type annotation.